test/unit-test/ota_utest.c

```diff
@@ -1643,6 +1674,27 @@ void test_OTA_ReceiveFileBlockCompleteDynamicBufferHttp()
     test_OTA_ReceiveFileBlockCompleteHttp();
 }
 
+/* Test that extractAndStoreArray fails if device does not have sufficient
+ * memory to allocate the string/array. */
+void test_OTA_ExtractArrayMemAllocFails()
+{
+    otaInterfaces.os.mem.malloc = mockMallocAlwaysFail;
+
+    memset( &pOtaAppBuffer, 0, sizeof( pOtaAppBuffer ) );
+    otaGoToState( OtaAgentStateWaitingForFileBlock );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
+}
+
+/* Test that extractAndStoreArray fails if user buffer is insufficient
+ * to allocate the string/array (here streamname). */
+void test_OTA_ExtractArrayInsufficientBuffer()
+{
+    pOtaAppBuffer.streamNameSize = 5;
```

Comment:
Where does the number 5 come from? Can you either:
1) Find a macro (something like min size) and base this number off of that
or
2) Add a comment about why 5? This would help us in the future to see if it needs to be updated based on library changes.

Reply:
Changed to macro OTA_INVALID_STREAM_NAME_SIZE and added comment